### PR TITLE
Update arrow using Plasma with glog

### PR DESCRIPTION
**Pull request by guoyuhong**
## What do these changes do?

1. Update Arrow to https://github.com/apache/arrow/pull/2522 which has glog in Plasma.
2. According to Arrow's recent change, https://github.com/apache/parquet-cpp is no longer used. We can disable this parquet building part and enable it in Arrow cmake.
## Related issue number
N/A

### CODE REVIEW

File: cmake/Modules/ArrowExternalProject.cmake [hunk 1]
```diff
@@ -53,11 +56,27 @@ set(ARROW_CMAKE_ARGS
...
+    -DARROW_WITH_ZLIB=ON)
+else()
+  set(ARROW_CMAKE_ARGS ${ARROW_CMAKE_ARGS}
+    -DARROW_WITH_THRIFT=OFF
+    -DARROW_PARQUET=OFF
+    -DARROW_WITH_ZLIB=OFF)
+endif ()
+if (APPLE)
+  set(ARROW_CMAKE_ARGS ${ARROW_CMAKE_ARGS}
+    -DBISON_EXECUTABLE=/usr/local/opt/bison/bin/bison)
```
Comment (raulchen) [inline cmake/Modules/ArrowExternalProject.cmake]: do not hard code this path, use the result of `which bison`

Comment (guoyuhong) [reply cmake/Modules/ArrowExternalProject.cmake]: I think the default bison path will cause CI failure. The version is not correct. That is why I added commit of "Fix Mac building error in CI". Actually, this hard code exists in many places of ray: [example1](https://github.com/ray-project/ray/blob/master/cmake/Modules/ParquetExternalProject.cmake#L35), [example2](https://github.com/ray-project/ray/blob/master/thirdparty/scripts/build_parquet.sh#L12).

Anyway, I can try it to see whether it works or not.

Comment (guoyuhong) [reply cmake/Modules/ArrowExternalProject.cmake]: @raulchen It failed with that, so I changed back.

**Comment by guoyuhong** [comment]
@pcmoritz Hi, Philipp, do you have time to take a look at this PR?

**Comment by pcmoritz** [comment]
Thanks, I also removed the corresponding line from setup.sh :)

**Review by pcmoritz — APPROVED** [review]
LGTM, can be merged if tests pass

**Pull request merged**